pyface/data_view/data_formats.py

```diff
@@ -0,0 +1,293 @@
+    """ Deserialize the first column of a CSV bytestring as a list.
+
+    Any columns beyond the first are ignored.
+
+    Parameters
+    ----------
+    raw_data : bytes
+        The serialized CSV data as a byte string.
+    delimiter : str
+        The CSV delimiter.
+    encoding : str
+        The encoding of the bytes
+    **kwargs
+        Additional arguments to csv.reader.
+
+    Returns
+    -------
+    data : list of str
+        The data extracted as a list of strings.
+    """
+    with StringIO(raw_data.decode(encoding)) as fp:
+        reader = csv.reader(fp, delimiter=delimiter, **kwargs)
+        data = [row[0] for row in reader]
+    return data
+
+
+text_row_format = DataFormat(
+    'text/plain',
+    partial(to_csv_row, delimiter='\t', lineterminator=os.linesep),
+    partial(from_csv_row, delimiter='\t'),
+)
+csv_row_format = DataFormat('text/csv', to_csv_row, from_csv_row)
+text_column_format = DataFormat(
+    'text/plain',
+    partial(to_csv_column, delimiter='\t', lineterminator=os.linesep),
+    partial(from_csv_column, delimiter='\t'),
+)
+csv_column_format = DataFormat('text/csv', to_csv_column, from_csv_column)
+
+
+# 2D formats
+
+def to_csv(data, delimiter=',', encoding='utf-8', **kwargs):
+    """ Serialize a list of lists to a CSV bytestring.
+
+    Parameters
+    ----------
+    data : list of lists
+        The data to be serialized.  Any elements which are not strings will
+        be converted to strings by calling ``str()``.
+    delimiter : str
+        The CSV delimiter.
+    encoding : str
+        The encoding of the bytes
+    **kwargs
+        Additional arguments to csv.writer.
+
+    Returns
+    -------
+    raw_data : bytes
+        The serialized data as a bytestring.
+    """
+    with StringIO() as fp:
+        writer = csv.writer(fp, delimiter=delimiter, **kwargs)
+        for row in data:
+            writer.writerow(row)
+        return fp.getvalue().encode(encoding)
+
+
+def from_csv(raw_data, delimiter=',', encoding='utf-8', **kwargs):
+    """ Deserialize a CSV bytestring.
+
+    Parameters
+    ----------
+    raw_data : bytes
+        The serialized CSV data as a byte string.
+    delimiter : str
+        The CSV delimiter.
+    encoding : str
+        The encoding of the bytes
+    **kwargs
+        Additional arguments to csv.reader.
+
+    Returns
+    -------
+    data : list of list of str
+        The data extracted as a list of lists of strings.
+    """
+    with StringIO(raw_data.decode(encoding)) as fp:
+        reader = csv.reader(fp, delimiter=delimiter, **kwargs)
+        data = list(reader)
+    return data
+
+
+def to_npy(data):
+    """ Serialize an array to a bytestring using .npy format.
+
+    Parameters
+    ----------
+    data : array-like
+        The array to be serialized.
+
+    Returns
+    -------
+    raw_data : bytes
+        The serialized data as a bytestring.
+    """
+    from numpy import save, atleast_2d
```

Comment:
Can we follow the usual way we do/use numpy imports here i.e. `import numpy as np; np.save/np.atleast_2d` ?